packages/ember-utils/lib/inspect.ts

```diff
@@ -16,45 +24,96 @@ const objectToString = Object.prototype.toString;
   @since 1.4.0
   @private
 */
-export default function inspect(obj: any | null | undefined): string {
-  if (obj === null) {
-    return 'null';
+export default function inspect(this: any, obj: any | null | undefined): string {
+  // detect Node util.inspect call inspect(depth: number, opts: object)
+  if (typeof obj === 'number' && arguments.length === 2) {
+    return this;
   }
-  if (obj === undefined) {
-    return 'undefined';
+
+  return inspectValue(obj, 0);
+}
+
+function inspectValue(value: any | null | undefined, depth: number, seen?: WeakSet<object>) {
+  let valueIsArray = false;
+  switch (typeof value) {
+    case 'undefined':
+      return 'undefined';
+    case 'object':
+      if (value === null) return 'null';
+      if (isArray(value)) {
+        valueIsArray = true;
+        break;
+      }
+      // is toString Object.prototype.toString or undefined then traverse
+      if (value.toString === objectToString || value.toString === undefined) {
+        break;
+      }
+      // custom toString
+      return value.toString();
+    case 'function':
+      return value.toString === functionToString
+        ? `function ${value.name}() { ... }`
+        : value.toString();
+    case 'string':
+      return stringify(value);
+    case 'symbol':
+    case 'boolean':
+    case 'number':
+    default:
+      return value.toString();
   }
-  if (Array.isArray(obj)) {
-    return `[${obj}]`;
+  if (seen === undefined) {
+    seen = new WeakSet();
```

Comment:
CI is failing in IE11 due to unguarded usage of WeakSet, can you update this to `import { WeakSet } from 'ember-utils'` instead?